MediaGallerySynchronization/Model/Synchronize.php

```diff
@@ -101,6 +113,8 @@ public function execute(): void
             }
         }
 
+        $this->resolveNonExistedAssets->execute($assetsPaths);
```

Comment:
Can you please introduce this operation in an instance of `SynchronizeFilesInterface` added to synchronizers pool. And utilize `\Magento\MediaGalleryApi\Api\DeleteAssetsByPathsInterface` for removing assets